### PR TITLE
[MM-28810] channel_store_categories: check if object is nil after the query

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Add a check where query returns `nil` and no `error`.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28810

### CODE REVIEW

File: store/sqlstore/channel_store_categories.go [hunk 1]
```diff
@@ -813,6 +814,8 @@ func (s SqlChannelStore) addChannelToFavoritesCategoryT(transaction *gorp.Transa
 	var channel *model.Channel
 	if obj, err := transaction.Get(&model.Channel{}, preference.Name); err != nil {
```
Comment (agnivade) [inline store/sqlstore/channel_store_categories.go:815]: I thought we get a `sql.ErrNoRows` if no rows match?

In any case, can we add a test ?

**Review by agnivade — CHANGES_REQUESTED** [review]

File: store/sqlstore/channel_store_categories.go (hunk 1 above)
Comment (isacikgoz) [reply store/sqlstore/channel_store_categories.go:815]: From [mattermost/gorp](https://github.com/mattermost/gorp/blob/master/gorp.go):
```golang
err = row.Scan(dest...)
if err != nil {
	if err == sql.ErrNoRows {
		err = nil
	}
	return nil, err
}
```

Anyway, added a couple of test for this.

Comment (agnivade) [reply store/sqlstore/channel_store_categories.go:815]: Oh .. this is only for `Get`. We should actually use `SelectOne`. Anyways, no need to change that now.

File: store/storetest/channel_store_categories.go [hunk 2]
```diff
@@ -1676,3 +1677,49 @@ func testDeleteSidebarCategory(t *testing.T, ss store.Store, s SqlSupplier) {
...
+		assert.Nil(t, nErr)
+
+		err := ss.Channel().UpdateSidebarChannelsByPreferences(&model.Preferences{
+			model.Preference{
+				Name:     "fakeid",
+				Category: model.PREFERENCE_CATEGORY_FAVORITE_CHANNEL,
+				Value:    "true",
+			},
+		})
+		if assert.NotNil(t, err) {
```
Comment (agnivade) [inline store/storetest/channel_store_categories.go:1720]: Should we instead check that a panic hasn't happened like in the other panic tests?

Comment (isacikgoz) [reply store/storetest/channel_store_categories.go:1720]: Yeah, makes sense.

**Review by agnivade — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]
Strange. I thought `Get` would return an error when no results were found. I guess it's not consistent with `SelectOne`.

**Comment by agnivade** [comment]
@hmhealey - Welcome to the wonderful world of gorp.

**Review by lindalumitchell — APPROVED** [review]

**Pull request merged**